DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -1,369 +1,424 @@
+        /// <returns></returns>
+        /// -----------------------------------------------------------------------------
+        protected override bool IsCorrectType(InstallFileType type)
+        {
+            return type == InstallFileType.Script;
+        }
+
+        /// -----------------------------------------------------------------------------
+        /// <summary>
+        /// The ProcessFile method determines what to do with parsed "file" node.
+        /// </summary>
+        /// <param name="file">The file represented by the node.</param>
+        /// <param name="nav">The XPathNavigator representing the node.</param>
+        /// -----------------------------------------------------------------------------
+        protected override void ProcessFile(InstallFile file, XPathNavigator nav)
+        {
+            string type = nav.GetAttribute("type", string.Empty);
+            if (file != null && this.IsCorrectType(file.Type) && this.IsValidScript(file.Name))
+            {
+                if (file.Name.StartsWith("install.", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    // This is the initial script when installing
+                    this._installScript = file;
+                }
+                else if (file.Name.StartsWith("preupgrade.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("preupgrade", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    this._preUpgradeScripts.Add(file);
+                }
+                else if (file.Name.StartsWith("upgrade.", StringComparison.InvariantCultureIgnoreCase)
+                    || file.Name.StartsWith("postupgrade.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("postupgrade", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    this._postUpgradeScripts.Add(file);
+                }
+                else if (type.Equals("install", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    // These are the Install/Upgrade scripts
+                    this.InstallScripts[file.Version] = file;
+                }
+                else if (file.Name.StartsWith("uninstall.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("uninstall", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    // These are the Uninstall scripts
+                    this.UnInstallScripts[file.Version] = file;
+                }
+                else
+                {
+                    // we couldn't determine the file type
+                    this.Log.AddFailure(string.Format(Util.SQL_Manifest_BadFile, file.Name));
+                    return;
+                }
+            }
+            else
+            {
+                // bad file
+                this.Log.AddFailure(Util.SQL_Manifest_Error);

```

Comment:
Should this failure reference the file name, too?

Reply:
The file is potentially null in the logic above